lib/lago_mcp_client/lago_mcp_client/mistral/client.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+module LagoMcpClient
+  module Mistral
+    class Client
+      MISTRAL_CONVERSATIONS_URL = "https://api.mistral.ai/v1/conversations"
+
+      def start_conversation(inputs:, &block)
+        payload = {
+          agent_id: ENV["MISTRAL_AGENT_ID"],
+          inputs: normalize_inputs(inputs),
+          stream: true
+        }
+        stream_conversation(payload, MISTRAL_CONVERSATIONS_URL, &block)
+      end
+
+      def append_to_conversation(conversation_id:, inputs:, &block)
+        url = "#{MISTRAL_CONVERSATIONS_URL}/#{conversation_id}"
+        payload = {inputs: inputs, stream: true}
+        stream_conversation(payload, url, &block)
+      end
+
+      private
+
+      def normalize_inputs(inputs)
+        if inputs.is_a?(String)
+          [{role: "user", content: inputs}]
+        else
+          inputs
+        end
+      end
+
+      def stream_conversation(payload, url)
+        http_client = LagoHttpClient::Client.new(url, read_timeout: 120)
+        headers = {
+          "Authorization" => "Bearer #{ENV["MISTRAL_API_KEY"]}",
+          "Accept" => "text/event-stream"
+        }
+
+        conversation_id = nil
+        outputs = []
+        tool_calls = []
+
+        http_client.post_with_stream(payload, headers) do |_type, data, _id, _reconnection_time|
+          next if data == "[DONE]"
```

Comment:
Just wondering if this block could be extracted into a proper "ConversationHandler" class that would be responsible from parsing the messages and acting accordingly.
It could be done in similar way of what we are doing with Webhook handler while keeping the conversation context (outputs, tool_calls...)